### PR TITLE
feat: 773 - new method "getProductNameBrandQuantity"

### PR DESCRIPTION
**Pull request by monsieurtanuki**
### What
- New method `getProductNameBrandQuantity`.
- Added needed "abbreviated_product_name" product fields.

### Fixes bug(s)
- Closes: #773

### Impacted files
* `api_get_product_test.dart`: tested "abbreviated product name" fields
* `product.dart`: added method `getProductNameBrandQuantity`; added "abbreviated product name" fields
* `product.g.dart`: generated
* `product_fields.dart`: added "abbreviated product name" fields

### CODE REVIEW

File: lib/src/model/product.dart [hunk 1]
```diff
@@ -925,4 +943,75 @@ class Product extends JsonObject {
     }
     _nutriments = nutriments;
   }
+
+  /// Returns the best version of a product name.
+  ///
+  /// cf. openfoodfacts-server/lib/ProductOpener/Products.pm
+  String getBestProductName(final OpenFoodFactsLanguage language) {
```
Comment (g123k) [inline lib/src/model/product.dart:950]: I would prefer to return a `null` value if there is no match; otherwise you force checking an empty string

**Comment by monsieurtanuki** [comment]
Closing as stale.

File: lib/src/model/product.dart [hunk 2]
```diff
@@ -951,4 +969,75 @@ class Product extends JsonObject {
...
+    }
+    if ((tmp = genericName)?.isNotEmpty == true) {
+      return tmp!;
+    }
+    if ((tmp = abbreviatedNameInLanguages?[language])?.isNotEmpty == true) {
+      return tmp!;
+    }
+    if ((tmp = abbreviatedName)?.isNotEmpty == true) {
+      return tmp!;
+    }
```
Comment (M123-dev) [inline lib/src/model/product.dart:995]: Can't we do a 

tmp ??= x;
tmp ??= y;

return tmp;

**Review by M123-dev — APPROVED** [review]
First of all sorry for my very irregular contributions, bout to write my finals in April and Mai, guess especially this PR somehow got  lost in the notifications. 

Added a small comment, otherwise good. Especially the new bestName method could be great for outside package users

File: lib/src/model/product.dart (hunk 2 above)
Comment (monsieurtanuki) [reply lib/src/model/product.dart:995]: @M123-dev Looks tempting but we check "not null and not empty", not just "not null", therefore your suggestion would potentially return empty strings.

**Comment by monsieurtanuki** [comment]
Thank you @M123-dev for your review, and good luck for your finals!

**Pull request merged**